cmd/metal-api/internal/service/machine-service.go

```diff
@@ -1155,37 +1155,48 @@ func gatherNetworksFromSpec(ds *datastore.RethinkStore, allocationSpec *machineA
 		return nil, fmt.Errorf("given network ids are not unique")
 	}
 
-	if privateNetwork == nil {
+	if len(privateNetworks) == 0 {
 		return nil, fmt.Errorf("no private network given")
 	}
 
-	if privateNetwork.network.ProjectID != allocationSpec.ProjectID {
+	if primaryPrivateNetwork.network.ProjectID != allocationSpec.ProjectID {
 		return nil, fmt.Errorf("the given private network does not belong to the project, which is not allowed")
 	}
 
-	for _, ipString := range allocationSpec.IPs {
-		ip, err := ds.FindIPByID(ipString)
-		if err != nil {
-			return nil, err
+	for _, pn := range privateNetworks {
+		if pn.network.PartitionID != partitionPrivateSuperNetwork.PartitionID {
+			return nil, fmt.Errorf("private networks must be in the partition where the machine is going to be placed")
```

Comment:
the constraint "all private networks must be located in the same partition" is enforced here.